src/fsharp/TypeChecker.fs

```diff
@@ -4224,10 +4224,55 @@ let rec TcTyparConstraint ridx cenv newOk checkCxs occ (env: TcEnv) tpenv c =
     | WhereTyparSubtypeOfType(tp, ty, m) ->
         let ty', tpenv = TcTypeAndRecover cenv newOk checkCxs ItemOccurence.UseInType env tpenv ty
         let tp', tpenv = TcTypar cenv env newOk tpenv tp
-        if newOk = NoNewTypars && isSealedTy cenv.g ty' then 
-            errorR(Error(FSComp.SR.tcInvalidConstraintTypeSealed(), m))
-        AddCxTypeMustSubsumeType ContextInfo.NoContext env.DisplayEnv cenv.css m NoTrace ty' (mkTyparTy tp') 
+        let tryGetInlineInterface =
+            match ty' with
+            | TType_app(possibleInterface, _) ->
+                match possibleInterface.TryDeref with
+                | VSome interface' when interface'.IsFSharpInlineInterfaceTycon -> 
+                    Some interface'
+                | _ -> None
+            | _ -> None
+        match tryGetInlineInterface with
+        | Some interface' ->
+            let rec collectArgumentTypes ttype args =
+                match ttype with
+                | TType_fun(arg0, arg1) -> 
+                    let args0 = collectArgumentTypes arg0 []
+                    let args1 = collectArgumentTypes arg1 []
+                    args0 @ args1
+                | TType_tuple(_, args) ->
+                    args |> List.collect (fun arg -> collectArgumentTypes arg [])
+                | TType_app (_,_) as r -> args @ [r]
+                | TType_var _ as r -> args @ [r]
+                | _ -> args
+            for member' in interface'.MembersOfFSharpTyconSorted do
+                match member'.Type with
+                | TType_fun(_, args)->
```

Comment:
Likewise here.  (I'm not totally sure what this is doing as I'm taking a quick glance but I suspect there are existing helpers to do the sort of thing you're doing here)

Reply:
I'll have a look whether there's a fitting one.